java/wms-layer-url/src/main/java/com/esri/arcgisruntime/sample/wmslayerurl/MainActivity.java

```diff
@@ -0,0 +1,68 @@
+/* Copyright 2017 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.wmslayerurl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+
+import com.esri.arcgisruntime.layers.WmsLayer;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class MainActivity extends AppCompatActivity {
+
+    private MapView mMapView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // inflate MapView from layout
+        mMapView = findViewById(R.id.mapView);
+        // create a map with the BasemapType topographic
+        ArcGISMap map = new ArcGISMap(Basemap.Type.IMAGERY, 2.0, 18.0, 3);
+        // set the map to be displayed in this view
+        mMapView.setMap(map);
+
+        // hold a list of uniquely-identifying WMS layer names to display
+        List<String> wmsLayerNames = new ArrayList<>();
+        wmsLayerNames.add("0");
+
+        // create a new WMS layer displaying the specified layers from the service
+        WmsLayer wmsLayer = new WmsLayer(getString(R.string.wms_layer_url), wmsLayerNames);
```

Comment:
This service does not seem to be working currently - cant access in a browser, and loading layer fails with "invalid XML" error.

Reply:
It's an external service hosted by the USGS, which is intentional, which means it may occasionally go down. This is meant to show support for non ESRI managed WMSs.

This does mean if USGS changes the url or closes the service the sample won't work though...